src/webhook/validation/oneagent.go

```diff
@@ -90,3 +93,46 @@ func hasConflictingMatchLabels(labelMap, otherLabelMap map[string]string) bool {
 	otherLabelSelectorLabels := labels.Set(otherLabelMap)
 	return labelSelector.Matches(otherLabelSelectorLabels) || otherLabelSelector.Matches(labelSelectorLabels)
 }
+
+func getOneAgentVolumeStorageEnvVar(env []corev1.EnvVar) bool {
+	envVar := kubeobjects.FindEnvVar(env, "ONEAGENT_ENABLE_VOLUME_STORAGE")
+	if envVar != nil {
+		return envVar.Value == "true"
+	}
+	return false
+}
+
+func hasOneAgentVolumeStorageEnabled(dynakube *dynatracev1beta1.DynaKube) bool {
```

Comment:
the `env` access could be moved to `properties.go` and tested there. That could also simplify test and only require the `HostInjectSpec` to be tested 🙂

Reply:
Not sure if I understood that correctly, but the tests in `validation` test the validation routines not the `HostInejctSpec`. And those routines work based on the feature-flag and the env var as an input. Thus, testing the validation routines always requires a full dynakube as input. So how would moving the env var access to properties simplify tests here?